packages/itwinui-react/src/core/Menu/MenuItem.tsx

```diff
@@ -159,92 +206,129 @@ export const MenuItem = React.forwardRef((props, forwardedRef) => {
       case 'ArrowLeft': {
         if (parent.ref) {
           parent.ref.current?.focus();
+          parent.setIsSubmenuVisible(false);
           parent.setIsNestedSubmenuVisible(false);
         }
         event.stopPropagation();
         event.preventDefault();
         break;
       }
       case 'Escape': {
-        // focus might get lost if submenu closes so move it back to parent
-        parent.ref?.current?.focus();
+        dropdownMenuContext.close();
+
+        event.stopPropagation();
+        event.preventDefault();
```

Comment:
I think this whole branch might be unnecessary if we use [`bubbles`](https://floating-ui.com/docs/usedismiss#bubbles)

Reply:
Ohh, nice, hardcoded `bubbles: true` in `Popover.tsx` to get the same behavior as this branch. (https://github.com/iTwin/iTwinUI/pull/1942/commits/e6b628bd4cae8a3bd855b1107e31032faf378434).

Then in [`271c4bc`](https://github.com/iTwin/iTwinUI/pull/1942/commits/271c4bc791722d5acddba972c39c36e0e6c46465), I added a new prop to `usePopover` to add `bubbles: true` only for `MenuItem` and not for other uses of `usePopover`.

Reply:
The prop is not needed. Menu is the only place we use `FloatingTree`, so there are no "other uses". And even if we add `FloatingTree` in more places, I'd argue we want the bubbles behavior in all of those places too.

The one exception is tooltips. I don't think that dismissing a tooltip using <kbd>esc</kbd> should close other popovers. But since Tooltip has its own `useFloating` call, I believe this should already work fine.

Reply:
One more case I felt where `bubbles: true` was not working well was in `DatePickerInput`. Because <kbd>esc</kbd> was closing the entire filter instead of just the `DatePicker`.

<details>
<summary>bubbles: true/false</summary>

`bubbles: true`

https://github.com/iTwin/iTwinUI/assets/45748283/3041105d-0983-4f1d-967f-fbefbf01b9ba

`bubbles: false`

https://github.com/iTwin/iTwinUI/assets/45748283/4f3dedf6-1a6b-448b-9513-afc6f3820b7e

</details>

> The one exception is tooltips. I don't think that dismissing a tooltip using esc should close other popovers. But since Tooltip has its own `useFloating` call, I believe this should already work fine.

Yes, I just confirmed that `bubbles: true` does not affect the behavior when a tooltip (e.g. from an `IconButton`) is in a `Dialog`.

<details>
<summary>Screen-recording</summary>

https://github.com/iTwin/iTwinUI/assets/45748283/bd171f76-4967-4e42-aca0-48cc14c89151

</details>

Reply:
Oh interesting, in that case I think it should be dependent on whether there is a FloatingTree.

If usePopover called in a component within a FloatingTree, then it should bubble. If not, then don't bubble.

Reply:
Ohh, sounds good. Removed `triggerProps` and now only bubbling when inside a `<FloatingTree>` scope. (https://github.com/iTwin/iTwinUI/pull/1942/commits/5dac2d3aacf9ebec50c301cfa18e72857a30937d)